include/mpm_explicit.tcc

```diff
@@ -444,6 +452,40 @@ bool mpm::MPMExplicit<Tdim>::apply_nodal_tractions() {
   return status;
 }
 
+//! Assign materials to particle sets
+template <unsigned Tdim>
+bool mpm::MPMExplicit<Tdim>::assign_set_material() {
+  bool status = false;
+  // Assign material to particle sets
+  try {
+    // Get particle properties
+    auto particle_props = io_->json_object("particle");
+    // Get particle sets properties
+    auto particle_sets = particle_props["particle_sets"];
+    // Assign material to each particle sets
+    for (const auto psets : particle_sets) {
+      // Get set material id
+      auto set_material_id = psets["set_material_id"];
+      // Get set material from list of materials
+      auto set_material = materials_.at(set_material_id);
+      // Get sets ids
+      std::vector<mpm::Index> sids = psets["set_id"];
```

Comment:
Let's not create another copy here

Reply:
Is that mean using "  auto set_material = materials_.at(psets["material_id"]);   "?